pyartcd/pyartcd/pipelines/build_microshift.py

```diff
@@ -0,0 +1,164 @@
+
+            # Rebases and builds microshift
+            version, release = self.generate_microshift_version_release(release_name)
+            nvrs = await self._rebase_and_build_rpm(version, release)
+
+            # https://mirror.openshift.com/pockets/microshift/4.10-el8/stream/x86_64/os/
+
+            # Prints example schema
+            click.secho("Build completes. Please update the assembly schema in releases.yaml to pin the following NVR(s) to the assembly:\n", fg="green")
+            for nvr in nvrs:
+                click.secho(f"\t{nvr}", fg="green")
+            example_schema = yaml.safe_dump(self._generate_example_schema(nvrs))
+            click.secho(f"\nExample schema:\n\n{example_schema}", fg="green")
+
+            # Sends a slack message
+            # TODO: We might want this job to auto create a PR to ocp-build-data.
+            await self._slack_client.say(f"Hi @release-artists , microshift for assembly {self.assembly} has been successfully built.", slack_thread)
+            await self._slack_client.say(f"Please create a PR to pin the build:\n```\n{example_schema}\n```", slack_thread)
+        except Exception as err:
+            error_message = f"Error building microshift: {err}\n {traceback.format_exc()}"
+            self._logger.error(error_message)
+            await self._slack_client.say(error_message, slack_thread)
+            raise
+
+    @staticmethod
+    def generate_microshift_version_release(ocp_version: str, timestamp: Optional[str] = None):
+        if not timestamp:
+            timestamp = datetime.utcnow().strftime("%Y%m%d%H%M")
+        release_version = VersionInfo.parse(ocp_version)
+        version = f"{release_version.major}.{release_version.minor}.{release_version.patch}"
+        if release_version.prerelease is not None:
+            version += f"~{release_version.prerelease}"
+        release = f"{timestamp}.p?"
+        return version, release
+
+    async def _rebase_and_build_rpm(self, version, release: str) -> List[str]:
+        """ Rebase and build RPM
+        :param release: release field for rebase
+        :return: NVRs
+        """
+        cmd = [
+            "doozer",
+            "--group", self.group,
+            "--assembly", self.assembly,
+            "-r", "microshift",
+            "rpms:rebase-and-build",
+            "--version", version,
+            "--release", release,
+        ]
+        if self.runtime.dry_run:
+            cmd.append("--dry-run")
+        await exectools.cmd_assert_async(cmd, env=self._doozer_env_vars)
+
+        if self.runtime.dry_run:
+            return [f"microshift-{version}-{release.replace('.p?', '.p0')}.el8"]
+
+        # parse record.log
+        with open(Path(self._doozer_env_vars["DOOZER_WORKING_DIR"]) / "record.log", "r") as file:
+            record_log = parse_record_log(file)
+            return record_log["build_rpm"][-1]["nvrs"].split(",")
+
+    def _generate_example_schema(self, nvrs: List[str]) -> Dict:
```

Comment:
It would help to see an example schema in the docstrings for this function

Reply:
Done